Windows build env: download latest Win32OpenSSL

gregorydulin opened the pull request:
Added code to search Win32OpenSSL download page for the latest
Win32OpenSSL version, vs. hard-coding the version.

Depends on curl, grep, and sed (I *think* these are included with
Windows git bash)

gregorydulin committed:
Windows build env: download latest Win32OpenSSL

Added code to search Win32OpenSSL download page for the latest
Win32OpenSSL version, vs. hard-coding it.

Depends on curl, grep, and sed (I *think* these are included with git
bash)

mlyle commented:
this doesn't test it, but it's a required checklist item too

jenkins, test this please

gregorydulin commented:
I did some research on the wayback machine, and this will fail if there are two versions available with a lowercase letter in the version number (which was the case in May, at least).  I'll fix it.

gregorydulin committed:
make: tools: find latest Win32OpenSSL_1_0_*

Eliminates the need to modify make/tools.mk every time slproweb.com
updates their Win32OpenSSL packages

uses curl to get https://slproweb.com/prodcuts/Win32OpenSSL.html
uses grep to find lines with 'Win32OpenSSL-1_0_*'
uses head to grab the first matching line (in case they decide to make
two versions available)
uses sed to get the .exe's URL
